src/util/chooseRenormalize.ts

```diff
@@ -0,0 +1,64 @@
+function mean(values: number[]): number {
+  return values.reduce((memo: number, i: number) => memo + i) / values.length;
+}
+
+/**
+ * This defines a random number between 0 and "mass", which should cover the entire
+ * domain of the probability distribution. Then we will "walk" the probability mass
+ * until we exceed the generated value
+ */
+function basicChooseOne(probs: number[], mass: number): number {
+  const r = mass * Math.random();
```

Comment:
Where are we setting the seed for the random number generator?

Reply:
Sounds like this is part of a larger discussion

Reply:
from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
`Math.random()`:
>  The implementation selects the initial seed to the random number generation algorithm; it cannot be chosen or reset by the user.

There is another project that could be used to create a seeded PRNG: https://github.com/davidbau/seedrandom